scripts/debug_tools/reduce_repro_files.py

```diff
@@ -0,0 +1,439 @@
+    parser.add_argument(
+        '--clang_plugin_name',
+        default=None,
+        help="Name of the used clang plugin.")
+    parser.add_argument(
+        '--clang_plugin_path',
+        default=None,
+        help="Path to the used clang plugin.")
+    parser.add_argument(
+        '--repro_zip',
+        default=None,
+        required=True,
+        help='The zip which contains the files needed for reproduction '
+             '(generated by CodeChecker).')
+    parser.add_argument(
+        '-o', default='./bug_repro_dir',
+        help='The output dir which contains the reduced files '
+             'reproducing the bug.')
+    args = parser.parse_args()
+    # change the paths to absolute
+    repro_zip = os.path.abspath(args.repro_zip)
+    output_dir = os.path.abspath(args.o)
+
+    print('Reduce script running started.')
+    if args.clang is not None:
+        args.clang = os.path.abspath(args.clang)
+        os.environ['PATH'] = os.path.dirname(args.clang) + \
+            os.pathsep + os.environ['PATH']
+        if args.verbose:
+            print('using clang: ' + args.clang)
+    else:
+        args.clang = 'clang'
+        if args.verbose:
+            print('using clang from PATH')
+
+    pathOptions = prepare_all_cmd_for_ctu.PathOptions(
+        args.sources_root,
+        args.clang,
+        args.clang_plugin_name,
+        args.clang_plugin_path,
+        args.report_dir)
+
+    if args.verbose:
+        print("Creating output directory: " + output_dir)
+    try:
+        os.mkdir(output_dir)
+    except OSError:
+        print('error: Repro dir already exist!')
+        return
+
+    analyzer_command_file = os.path.join(output_dir, 'analyze.sh')
+
+    # unzip
+    os.chdir(os.path.dirname(repro_zip))
+    args.sources_root = os.path.abspath(args.sources_root)
+    if args.verbose:
+        print("Reducing test case from zip file: " + repro_zip)
+
+    for file_name in os.listdir(os.getcwd()):
+        if 'zip' in file_name or os.path.basename(output_dir) == file_name:
+            continue
+        if os.path.isfile(file_name):
+            os.remove(file_name)
+        else:
+            shutil.rmtree(file_name)
+
+    if args.verbose:
+        print('unzip reproduction files')
+    try:
+        output = subprocess.check_output(
+            ['timeout', '3', 'unzip', repro_zip], stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        print('error: unzip command has failed')
+        return
+
+    if args.verbose:
+        print('Preparing commands for CTU analysis.')
+    with nostdout():
+        prepare_all_cmd_for_ctu.prepare(pathOptions)
+
+    compile_cmds = json.load(open('./compile_cmd_DEBUG.json'))
+    for cmd in compile_cmds:
+        file_name = os.path.basename(cmd['file'])
+        if not isCppOrCFile(file_name):
+            continue
+        preproc = subprocess.Popen(
+            get_preprocess_cmd(
+                cmd['command'],
```

Comment:
This command fails if the compiler is not available on my machine on this path. However most of the times this is the case, since the bug is being reproduced on a machine other than the original checker machine.